backend/app.py

```diff
@@ -308,6 +314,138 @@ def create_machine():
+        change_message += " multimachine,"
+
+    # Case 5: paywall reported
+    paywall_new = True if request.args.get("paywall") == "true" else False
+    paywall_old = existing_machine_infos["properties"].get("paywall", False)
+    if paywall_new != paywall_old:
+        updated_machine_entry["properties"]["paywall"] = paywall_new
+        change_message += " paywall,"
+
+    # Case 6: address and / or location changed --> check for their correspondence
+    (lng_old, lat_old) = existing_machine_infos["geometry"]["coordinates"]
+    old_address = existing_machine_infos["properties"]["address"]
+    # if address or coordinates were changed, compare them and return warning if needed
+    address_okay = True
+    if latitude != lat_old or longitude != lng_old or address != old_address:
+        # Verify that address matches coordinates
+        found_coords, (lat, lng) = verify_machine_location(address, area, title)
+        # if address was changed but is not found (error only if address was changed)
+        if (not found_coords) and address != old_address:
+            return jsonify({"error": "Google Maps does not know this address"}), 400
+
+        dist = haversine((lat, lng), (latitude, longitude))
+        if dist > 1:  # km
+            address_okay = False  # triggers warning
+
+        # adapt dictionary entries
+        updated_machine_entry["properties"]["address"] = address
+        updated_machine_entry["properties"]["latitude"] = str(latitude)
+        updated_machine_entry["properties"]["longitude"] = str(longitude)
+        updated_machine_entry["geometry"]["coordinates"] = [longitude, latitude]
+        if address != old_address:
+            change_message += " address,"
+        if latitude != lat_old or longitude != lng_old:
+            change_message += " location,"
+
+    # replace or append to server_locations
+    if index_in_server_locations > 0:
+        server_locations["features"][index_in_server_locations] = updated_machine_entry
+    else:
+        server_locations["features"].append(updated_machine_entry)
+
+    # push to github
+    commit_message = (
+        f"request change to machine {machine_id} named {title}. {change_message[:-1]}"
+    )
+    commit_json_file(
+        server_locations,
+        DATA_BRANCH,
+        commit_message,
+        latest_commit_sha,
+        body=f"Request to change machine {machine_id} named {title}. {change_message[:-1]}",
+    )
+
+    ip_address = request.remote_addr
+    message_slack_raw(
+        ip=ip_address,
+        text=f"Request to change machine {machine_id} proposed: {change_message}",
+    )
+    if not address_okay:
+        return (
+            jsonify(
+                {
+                    "error": f"Change request submitted successfully. However, the address ({address}) seems >1km away from coordinates ({latitude}, {longitude}). Consider adjusting your edits such that coordinates and address are aligned."
```

Comment:
This is the warning that is shown when address does not match the coordinate location

Reply:
Makes sense. There might be edge cases where users cant change the address, but let's not worry right now